src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -1028,7 +1028,11 @@ private static void VerifyModelForOutVarInNotExecutableCode(SemanticModel model,
 
             var typeInfo = model.GetTypeInfo(decl);
             Assert.Equal(expectedType, typeInfo.Type);
-            Assert.Equal(expectedType, typeInfo.ConvertedType);
+            if (decl.IsOutDeclaration())
+            {
+                Assert.Equal(expectedType, typeInfo.ConvertedType);
```

Comment:
I am also not comfortable with simply not testing behavior. There are some expectations for each scenario, when they do not match the ```expectedType```, we could pass expectations via dedicated parameter and verify against that.  Then the syntax check wouldn't be needed at all.